spec/sec-properties-of-the-regexp-prototype-object-patch.html

```diff
@@ -153,24 +154,26 @@ <h1><ins>MakeGetIndices ( _S_ )</ins></h1>
         1. Let _S_ be _f_.[[InputString]].
         1. Let _indices_ be _f_.[[MatchIndices]].
         1. Assert: _indices_ is a List.
+        1. Let _groupNames_ be _f_.[[GroupNames]].
+        1. Assert: _groupNames_ is a List or is *undefined*.
         1. Let _n_ be the number of elements in _indices_.
         1. Assert: _n_ &lt; 2<sup>32</sup>-1.
         1. Set _A_ to ! ArrayCreate(_n_).
         1. Assert: The value of _A_'s `"length"` property is _n_.
-        1. If there exists an _element_ of _indices_ such that _element_ is a Match Record and _element_.[[GroupName]] is not *undefined*, then
+        1. If _groupNames_ is not *undefined*, then
           1. Let _groups_ be ! ObjectCreate(*null*).
         1. Else,
           1. Let _groups_ be *undefined*.
         1. Perform ! CreateDataProperty(_A_, `"groups"`, _groups_).
         1. For each integer _i_ such that _i_ &ge; 0 and _i_ &lt; _n_, do
-          1. Let _matchIndices_ be _i_<sup>th</sup> element of _indices_.
+          1. Let _matchIndices_ be _indices_[_i_].
```

Comment:
is this typically how items of Lists are extracted?

Reply:
Yes, it is explicitly called out in the specification types section. The ordinal representation (i.e. "3rd element of x") is 1-based while the indexed version (i.e. "x[2]") is 0-based.